blockstore/api.go

```diff
@@ -15,14 +16,22 @@ type ChainIO interface {
 }
 
 type apiBlockstore struct {
-	api ChainIO
+	api   ChainIO
+	cache *lru.Cache[cid.Cid, []byte]
 }
 
 // This blockstore is adapted in the constructor.
 var _ BasicBlockstore = (*apiBlockstore)(nil)
 
 func NewAPIBlockstore(cio ChainIO) Blockstore {
-	bs := &apiBlockstore{api: cio}
+	lc, err := lru.New[cid.Cid, []byte](1024)
```

Comment:
I recommend leaving an in-line comment about the rationale for picking this size.